src/init/pkg_installer.sh

```diff
@@ -129,86 +123,46 @@ if [ -z "${WAZUH_REVISION}" ] ; then
-[ -d "./var/selinux" ] && FOLDERS_TO_BACKUP+=(${WAZUH_HOME}/var/selinux)
-
-for dir in "${FOLDERS_TO_BACKUP[@]}"; do
-    mkdir -p "${TMP_DIR_BACKUP}${dir}"
-    rsync -a --exclude "diff" ${dir}/* "${TMP_DIR_BACKUP}${dir}"
-done
-
-if [ -f $OSSEC_INIT_FILE ]; then
-    mkdir -p "${TMP_DIR_BACKUP}/etc"
-    cp -p $OSSEC_INIT_FILE "${TMP_DIR_BACKUP}/etc"
-fi
-
 # Check if systemd is used
 # RHEL 8 >= services must must be installed in /usr/lib/systemd/system/
 if [ -f /usr/lib/systemd/system/${SERVICE}.service ] && [ ! -h /usr/lib/systemd/system ]; then
     SYSTEMD_SERVICE_UNIT_PATH=/usr/lib/systemd/system/${SERVICE}.service
-    mkdir -p "${TMP_DIR_BACKUP}/usr/lib/systemd/system/"
-    cp -a "${SYSTEMD_SERVICE_UNIT_PATH}" "${TMP_DIR_BACKUP}${SYSTEMD_SERVICE_UNIT_PATH}"
 fi
 # Others
 if [ -f /etc/systemd/system/${SERVICE}.service ] && [ ! -h /etc/systemd/system ]; then
     SYSTEMD_SERVICE_UNIT_PATH=/etc/systemd/system/${SERVICE}.service
-    mkdir -p "${TMP_DIR_BACKUP}/etc/systemd/system/"
-    cp -a "${SYSTEMD_SERVICE_UNIT_PATH}" "${TMP_DIR_BACKUP}${SYSTEMD_SERVICE_UNIT_PATH}"
 fi
 
 # Init backup
 # REHL <= 6 / Amazon linux
 if [ -f "/etc/rc.d/init.d/${SERVICE}" ] && [ ! -h /etc/rc.d/init.d ]; then
     INIT_PATH="/etc/rc.d/init.d/${SERVICE}"
-    mkdir -p "${TMP_DIR_BACKUP}/etc/rc.d/init.d/"
-    cp -a "${INIT_PATH}" "${TMP_DIR_BACKUP}${INIT_PATH}"
 fi
 
 if [ -f "/etc/init.d/${SERVICE}" ] && [ ! -h /etc/init.d ]; then
     INIT_PATH="/etc/init.d/${SERVICE}"
-    mkdir -p "${TMP_DIR_BACKUP}/etc/init.d/"
-    cp -a "${INIT_PATH}" "${TMP_DIR_BACKUP}${INIT_PATH}"
 fi
 
-# Saves modes and owners of the directories
-BACKUP_LIST_FILES=$(find "${TMP_DIR_BACKUP}/" -type d)
+# Generate Backup
+BDATE=$(date +"%m-%d-%Y_%H-%M-%S")
+mkdir -p ./backup
 
-while read -r line; do
-    org=$(echo "${line}" | awk "sub(\"${TMP_DIR_BACKUP}\",\"\")")
-    chown --reference=$org $line >> ./logs/upgrade.log 2>&1
-    chmod --reference=$org $line >> ./logs/upgrade.log 2>&1
-done <<< "$BACKUP_LIST_FILES"
+echo "$(date +"%Y/%m/%d %H:%M:%S") - Generating Backup." >> ./logs/upgrade.log
 
+FOLDERS_TO_BACKUP=($WAZUH_HOME/{active-response,bin,etc,lib,queue,ruleset,wodles,agentless,logs/{ossec,wazuh},var/selinux} \
+                   $OSSEC_INIT_FILE \
+                   $SYSTEMD_SERVICE_UNIT_PATH \
+                   $INIT_PATH)
+FOLDERS_TO_EXCLUDE=($WAZUH_HOME/{queue/diff,logs/upgrade.log})
 
-# Generate Backup
-mkdir -p ./backup
-if [ "$(ls -A ${TMP_DIR_BACKUP})" ]; then
-    tar czf ./backup/backup_[${BDATE}].tar.gz -C ${TMP_DIR_BACKUP} . >> ./logs/upgrade.log 2>&1
-else
-    echo "$(date +"%Y/%m/%d %H:%M:%S") - Nothing to compress while creating the Backup, interrupting upgrade." >> ./logs/upgrade.log
-    exit 1
-fi
+for i in ${FOLDERS_TO_BACKUP[@]}
+do
+    [ -e $i ] && echo $i
+done | xargs tar -C / --exclude=$FOLDERS_TO_EXCLUDE -zcvf ./backup/backup_[${BDATE}].tar.gz >> ./logs/upgrade.log 2>&1
```

Comment:
`$FOLDERS_TO_EXCLUDE` is a list. This results into:
```
--exclude /var/ossec/queue/diff
```

But _logs/upgrade.log_ does not apply. We need something like:

```
--exclude /var/ossec/queue/diff  --exclude /var/ossec/logs/upgrade.log
```